rpc/types/src/macros.rs

```diff
@@ -269,9 +323,34 @@ macro_rules! define_request_and_response_doc {
             "), [",
             $request_or_response,
             "](",
-            stringify!($type_name),
+            stringify!($request_or_response_type),
+            ")."
+        )
+    };
+}
+pub(crate) use __define_request_and_response_doc;
+
+//---------------------------------------------------------------------------------------------------- Macro
+/// Output a string link to `monerod` source code.
+macro_rules! monero_definition_link {
+    (
+        $commit:ident, // Git commit hash
+        $file_path:literal, // File path within `monerod`'s `src/`, e.g. `rpc/core_rpc_server_commands_defs.h`
+        $start:literal$(..=$end:literal)? // File lines, e.g. `0..=123` or `0`
+    ) => {
+        concat!(
+            "[Definition](https://github.com/monero-project/monero/blob/",
+            stringify!($commit),
+            "/src/",
+            $file_path,
+            "#L",
+            stringify!($start),
+            $(
+                "-L",
+                stringify!($end),
+            )?
             ")."
         )
     };
 }
```

Comment:
This documents the original definition for pretty much every type in `cuprate-rpc-types`.